typing/typeclass.ml

```diff
@@ -724,7 +731,7 @@ and class_field_aux self_loc cl_num self_type meths vars
                     let sty = Ast_helper.Typ.force_poly sty in
                     let cty' = Typetexp.transl_simple_type val_env false sty in
                     let ty' = cty'.ctyp_type in
-              Ctype.unify val_env ty' ty
+              Ctype.unify class_env.val_env ty' ty
```

Comment:
Unless I am mistaken, there is `val_env` in scope and it is equal to `class_env.val_env` here, so it could be used instead.